### PR TITLE
Decompile z_quake.c and code_800D31A0.c

### PR DESCRIPTION
**Pull request by Random06457**
- `code_800D31A0.c` : OK
- `z_quake.c` : OK

### CODE REVIEW

File: include/z64.h [hunk 1]
```diff
@@ -1277,4 +1277,47 @@ typedef struct {
     /* 0x10 */ OSTime resetTime;
 } PreNmiBuff; // size = 0x18 (actually osAppNmiBuffer is 0x40 bytes large but the rest is unused)
 
+typedef struct
+{
```
Comment (Roman971) [inline include/z64.h]: Brace style is incorrect here and in other structs added.

File: src/code/code_800D31A0.c [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#include <global.h>
+#include <padmgr.h>
+
+extern PadMgr gPadMgr;
+
+u32 D_8012DBC0 = 0;
+
+void func_800D31A0() {
+    osSyncPrintf("\x1b[31m\n**** Freeze!! ****\n\x1b[m");
```
Comment (Roman971) [inline src/code/code_800D31A0.c]: VT macros should be used here.

File: src/code/code_800D31A0.c [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    while (true) {
+        func_800FF4AC(1000); // msleep
+    }
+}
+
+void func_800D31F0() {
+    PadMgr* padMgr = (PadMgr*)(u32)&gPadMgr; // cast required to match
+
+    D_8012DBC0 = (padMgr->unk_2A8 & 2) != 0;
```
Comment (Roman971) [inline src/code/code_800D31A0.c]: I suggest using `!!` if it still matches.
```suggestion
    D_8012DBC0 = !!(padMgr->unk_2A8 & 2);
```

File: src/code/code_800D31A0.c [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#include <global.h>
+#include <padmgr.h>
+
+extern PadMgr gPadMgr;
+
+u32 D_8012DBC0 = 0;
```
Comment (Roman971) [inline src/code/code_800D31A0.c]:
```suggestion
u32 D_8012DBC0 = false;
```

File: src/code/code_800D31A0.c (hunk 3 above)
Comment (Random06457) [reply src/code/code_800D31A0.c]: `!!(padMgr->unk_2A8 & 2)` causes a regalloc issue

**Pull request merged**